### PR TITLE
View document

### PR DESCRIPTION
**Pull request by apetcu**

### CODE REVIEW

File: src/app/core/store/effects/documents.effect.ts [hunk 1]
```diff
@@ -22,6 +22,17 @@ export class DocumentsEffect {
           )
       ));
 
+  @Effect()
+  loadConsolidatedDocument$ = this.actions$.pipe(
+      ofType(documentsActions.DocumentsActionTypes.LoadConsolidatedDocumentById),
+      map((action: documentsActions.LoadConsolidatedDocumentById) => action.payload),
+      switchMap((entry) => this.documentsService.getConsolidated(entry)
+          .pipe(
+              map((documentsPage: DocumentConsolidate) => new documentsActions.LoadConsolidatedDocumentByIdSuccess(documentsPage)),
```
Comment (bogdanconstantinescu) [inline src/app/core/store/effects/documents.effect.ts]: It's a bad idea to keep instances of objects in the redux store. I'd recommend using a JSON representation of the object inside the store and hydrate that to an actual object in the selectors.

File: src/app/core/store/reducers/documents.reducer.ts [hunk 2]
```diff
@@ -45,6 +46,24 @@ export function reducer(state = initialState, action: fromDocuments.DocumentsAct
...
       };
     }
+
+    case fromDocuments.DocumentsActionTypes.LoadConsolidatedDocumentByIdSuccess: {
+      console.log(action.payload);
+      const theDocument: DocumentConsolidate = action.payload;
+      const entities = {
+        ...state.entities,
+        [theDocument.id]: {
+          ...state.entities[theDocument.id], consolidated: theDocument
```
Comment (bogdanconstantinescu) [inline src/app/core/store/reducers/documents.reducer.ts]: Same things here, a JSON representation should be stored and hydrated back to an object in the selectors.

File: src/app/documents/documents.module.ts [hunk 3]
```diff
@@ -8,16 +8,22 @@ import * as fromComponents from './components';
 import * as fromContainer from './containers';
 import * as fromGuards from './guards';
 import { CommonModule } from '@angular/common';
+import { DocumentParagraphComponent } from './components/document-paragraph/document-paragraph.component';
+import { MatGridListModule } from '@angular/material';
+import { DocumentSidenavComponent } from './components/document-sidenav/document-sidenav.component';
```
Comment (bogdanconstantinescu) [inline src/app/documents/documents.module.ts]: No. You have barrel files where you have declared already the component (`components/index.ts`) - this is not needed

File: src/app/documents/documents.module.ts [hunk 4]
```diff
@@ -8,16 +8,22 @@ import * as fromComponents from './components';
...
     CommonModule,
     SharedModule,
     DocumentsRoutingModule,
+    MatGridListModule,
   ],
   declarations: [
     ...fromContainer.components,
     ...fromComponents.components,
+    DocumentParagraphComponent,
+    DocumentSidenavComponent,
```
Comment (bogdanconstantinescu) [inline src/app/documents/documents.module.ts]: This needs to go as well.

File: src/app/documents/documents.module.ts [hunk 5]
```diff
@@ -8,16 +8,22 @@ import * as fromComponents from './components';
...
+import { DocumentParagraphComponent } from './components/document-paragraph/document-paragraph.component';
+import { MatGridListModule } from '@angular/material';
+import { DocumentSidenavComponent } from './components/document-sidenav/document-sidenav.component';
 
 @NgModule({
   imports: [
     CommonModule,
     SharedModule,
     DocumentsRoutingModule,
+    MatGridListModule,
```
Comment (bogdanconstantinescu) [inline src/app/documents/documents.module.ts]: This needs to be moved to its proper place, `app/material.module.ts`

File: src/app/documents/guards/document-breakdown.guard.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  hasConsolidatedDetails(documentId: string): Observable<boolean> {
+    return this.store.pipe(select(fromStore.hasConsolidatedInformation(documentId)))
+        .pipe(
+            tap((hasDetail: boolean) => {
+              if (!hasDetail) {
+                return this.store.dispatch(new fromStore.LoadConsolidatedDocumentById(documentId));
+              }
+            })
```
Comment (bogdanconstantinescu) [inline src/app/documents/guards/document-breakdown.guard.ts]: This doesn't wait for the document to load, it just fires the loading and that's it.
If you want it to "wait" until the document is loaded you need an extra `filter(hasDetail => hasDetail)`.

**Comment by RaduCStefanescu** [comment]
@apetcu would you be able to have a look on the comments added to this PR? 
Thank you!

**Comment by georgebejan** [comment]
This is solved by #64

**Pull request closed without merging**